ztoc/ztoc_marshaler.go

```diff
@@ -132,30 +137,32 @@ func ztocToFlatbuffer(ztoc *Ztoc) (fb []byte, err error) {
 		}
 	}()
 
+	// ztoc - metadata
```

Comment:
I'm mostly onboard with these changes, but we're referring the the ztoc version as `ztoc - metadata` but `ztoc.Metadata` is actually FileMetadata, not metadata about the ztoc itself. Does it make sense to rename `TOC.Metadata` to `TOC.FileMetadata`?

Reply:
Changed to `FileMetadatas` since it's a slice and also avoid to having the field and type having the same name :)

Reply:
just FYI, this does change Ztoc's API in our library, whereas the previous change won't (both `ztoc.TOC.Metadata` and `ztoc.Metadata` work).

It's okay as long as no one uses soci as a dependency or uses `Ztoc.TOC.Metadata` directly.

Reply:
Is there a concern with the name and type being the same? This is a pretty common pattern in go.

Reply:
No concern, it's just this is a slice of `FileMetadata`, so I changed to plural `FileMetadatas`

Reply:
changed to `FileMetadata`